code/modules/mob/living/living_defense.dm

```diff
@@ -91,43 +91,77 @@
 /mob/living/proc/is_ears_covered()
 	return null
 
-/mob/living/bullet_act(obj/projectile/hitting_projectile, def_zone, piercing_hit = FALSE)
+/mob/living/bullet_act(obj/projectile/proj, def_zone, piercing_hit = FALSE)
 	. = ..()
-	if(. != BULLET_ACT_HIT)
-		return .
-	if(!hitting_projectile.is_hostile_projectile())
-		return BULLET_ACT_HIT
+	var/blocked = check_projectile_armor(def_zone, proj)
 
-	// we need a second, silent armor check to actually know how much to reduce damage taken, as opposed to
-	// on [/atom/proc/bullet_act] where it's just to pass it to the projectile's on_hit().
-	var/armor_check = check_projectile_armor(def_zone, hitting_projectile, is_silent = TRUE)
```

Comment:
Why is this no longer silent, won't this cause double armor messages?

Reply:
Actually should we just pass blocked as an argument to `bullet_act` and then give it a default value of `blocked = check_projectile_armor(def_zone, hitting_projectile)`? So we don't *need* two armor checks anyways?